src/bespokelabs/curator/request_processor/batch/anthropic_batch_request_processor.py

```diff
@@ -0,0 +1,224 @@
+        self.web_dashboard = "https://console.anthropic.com/settings/workspaces/default/batches"
+
+    @property
+    def max_requests_per_batch(self) -> int:
+        return 100_000
+
+    @property
+    def max_bytes_per_batch(self) -> int:
+        return 256 * 1024 * 1024  # 256 MB
+
+    @property
+    def max_concurrent_batch_operations(self) -> int:
+        return 100
+
+    def parse_api_specific_request_counts(
+        self, request_counts: MessageBatchRequestCounts
+    ) -> GenericBatchRequestCounts:
+        """
+        https://github.com/anthropics/anthropic-sdk-python/blob/e7c5fd1cf9226d73122870d07906664696da3ab8/src/anthropic/types/beta/messages/beta_message_batch_request_counts.py#L9
+        Request Counts (Anthropic): "processing", "canceled", "errored", "expired", "succeeded"
+        """
+        failed = request_counts.canceled + request_counts.errored + request_counts.expired
+        succeeded = request_counts.succeeded
+        processing = request_counts.processing
+        return GenericBatchRequestCounts(
+            failed=failed,
+            succeeded=succeeded,
+            total=processing + succeeded + failed,
+            raw_request_counts_object=request_counts.model_dump(),
+        )
+
+    def parse_api_specific_batch_object(
+        self, batch: MessageBatch, request_file: str | None = None
+    ) -> GenericBatch:
+        """
+        https://github.com/anthropics/anthropic-sdk-python/blob/e7c5fd1cf9226d73122870d07906664696da3ab8/src/anthropic/types/beta/messages/beta_message_batch.py#L53
+        Batch Status (Anthropic): "in_progress", "canceling", "ended"
+
+        https://github.com/anthropics/anthropic-sdk-python/blob/e7c5fd1cf9226d73122870d07906664696da3ab8/src/anthropic/types/beta/messages/beta_message_batch.py#L20-L51
+        Timing (Anthropic): "created_at", "cancel_initiated_at", "archived_at", "ended_at", "expires_at"
+        """
+        if batch.processing_status in ["cancelling", "in_progress"]:
+            status = "submitted"
+        elif batch.processing_status in ["ended"]:
+            status = "finished"
+        else:
+            raise ValueError(f"Unknown batch status: {batch.processing_status}")
+
+        return GenericBatch(
+            request_file=request_file,
+            id=batch.id,
+            created_at=batch.created_at,
+            finished_at=batch.ended_at,
+            status=status,
+            api_key_suffix=self.client.api_key[-4:],
+            request_counts=self.parse_api_specific_request_counts(batch.request_counts),
+            raw_batch=batch.model_dump(),
+            raw_status=batch.processing_status,
+        )
+
+    def create_api_specific_request_batch(self, generic_request: GenericRequest) -> dict:
+        if generic_request.response_format:
+            # TODO(Ryan) how can we support this the way litellm does?
+            raise NotImplementedError("response_format is not yet supported for Anthropic")
```

Comment:
Oh this isn't supported by Anthropic?

Reply:
This is, we just need to implement it for batch mode, since we are relying on litellm and instructor for doing it for online. We can maybe use their functions to help with this.

Reply:
https://github.com/bespokelabsai/curator/issues/225#issuecomment-2564378840

Looked into it a bit, here are the options ^

Reply:
Done